### ISSUE TITLE
Substitute combinations of blocks for single ones within ScratchAddons.

### ISSUE THREAD

**Issue by DroneBetter** [ScratchAddons/ScratchAddons#2132]
Combinations like `repeat until (not(boolean))` and `replace (item) of (list) with (((item) of (list))+(amount))` could be `while (boolean)` and `change (item) of (list) by (amount)`, respectively. An extension could replace these in the editor and allow creation of these blocks while executing them and reporting them to the Scratch servers as the original combinations, making them faster to program and more concise to read.

**Comment by RedGuy12** [comment]
While is a hacked block. We aren’t going to add hacked blocks.

So kind of like terser (https://github.com/terser/terser) for scratch

**Comment by WorldLanguages** [comment]
Yeah, this isn't a new concept, but it would still be complex to achieve properly. TBH I'd like to leave this challenge for editor forks like TurboWarp, and if they succeed, build an addon that shows the originally used blocks inside Scratch, as well as being able to use those without a third party editor.

**Comment by DroneBetter** [comment]
> While is a hacked block. We aren’t going to add hacked blocks.

It wouldn't add hacked blocks, only display existing ones to the user as whiles and replace all user-created whiles with backend repeat until nots, anyone without it installed would see the regular ones.
